datetime_tz/pytz_abbr.py

```diff
@@ -51,15 +51,51 @@
 class tzabbr(datetime.tzinfo):
   """A timezone abbreviation.
 
-  *WARNING*: This is not a tzinfo implementation! Trying to use this as tzinfo
-  object will result in failure.  We inherit from datetime.tzinfo so we can get
-  through the dateutil checks.
+  *WARNING*: This is a proxy object for an underlying `pytz` object only
+  intended to allow dateutil to probe the localized datetime for information.
+  datetime_tz's parsing logic will replace it with the relevant pytz zone
+  before returning to the end user.
   """
-  pass
+
+  def __init__(self, abbr, name, region, zone, dst):
+    self.abbr = abbr
+    self.name = name
+    self.region = region
+
+    if not isinstance(zone, pytz.tzinfo.BaseTzInfo):
+      zone = pytz.timezone(zone)
```

Comment:
`pytz.timezone(zone)` calls `zone.upper()` and fails.  It seems to be expecting a str?  not a `pytz.tzinfo.BaseTzInfo`

  https://github.com/stub42/pytz/blob/master/src/pytz/__init__.py#L163

Reply:
I you've got a `not` here... when it isn't one of those, it appears to be `pytz.UTC`.. which in older versions of `pytz` (~2017.2 for example) is a `datetime.tzinfo` rather than `pytz.tzinfo.BaseTzInfo`.

Reply:
Oh interesting, I didn't try it with old versions of `pytz`. Wow, I should really be kicking myself for removing the "test with all versions of pytz" test. I can just check if it's a string.